feat: saleor app integration

Ang-m4 opened the pull request:
## Description

This PR introduces the initial integration of `saleor-app` into the `platform-plugin-saleor`. It includes the following components:

- Creation of the app manifest and webhook setup.
- Base views for `<manifest>` and `<register>` endpoints.
- Definition of webhook handlers and URL paths.

## Webhooks Implemented

- **Order Updated**: Triggered whenever there is a change in the order, such as updates to the billing address, payment status (paid/unpaid), or fulfillment details.
- **Order Paid**: Triggered when a payment (full or partial) is received for an order.

## Payload Definition

Saleor allows customization of the webhook payload by selecting specific order fields. For both webhooks implemented in this PR, the following GraphQL query is used to fetch basic order and user information. For more details on available fields, refer to the [Order object documentation](https://docs.saleor.io/api-reference/orders/objects/order).

```graphql
order {
        id
        number
        status
        isPaid
        lines {
            id
            variant {
                name
                product {
                    name
                    externalReference
                }
            }
        }
        user { id, email }
    }
```

## How to Test

Once your environment is fully up and running and the plugin is installed, follow these steps:

1. Open your **Saleor Dashboard** and click on the **"Extensions"** option in the left-hand panel.
2. In the top-right corner, click **"Install External App"**.
3. Paste the manifest URL and wait for the app to install:  
   `http://local.overhang.io:8000/saleor/api/manifest`

   ![image](https://github.com/user-attachments/assets/8428d914-e7b9-4566-876d-5b12f26949e2)

4. Once installed, the app should appear as shown below:

   ![image](https://github.com/user-attachments/assets/105b2ead-43ce-407a-a876-fd790db1bc42)

5. Ensure that a **customer** and at least one **product** are created in your store.
6. Create an **order** using the customer and products, then click **"Finalize"** to complete the order.

   ![image](https://github.com/user-attachments/assets/00c37a44-8c7e-4cfe-bf59-5e801290138b)

7. Trigger the `ORDER_FULLY_PAID` webhook by:
   - Marking the order as **paid**
  
   ![image](https://github.com/user-attachments/assets/14de9b9a-e10d-4ee1-8956-c55c99c7286f)

8. This should trigger the webhook. Check your **LMS logs**  and verify the course enrollments if the user is enrolled correctly.

9. verify the orders fulfillment at the Saleor dashboard:
![image](https://github.com/user-attachments/assets/e95e28cb-401e-47b1-94f1-1fa2141d1547)


**Considerations:** 
- The customer email you use to create the order should match with a current LMS user.
- You will need to load the LMS courses to saleor again since the new `ExternalReference` attribute was added.
- The event will be triggered ONLY when you mark the order as PAID.

Ang-m4 committed:
feat: saleor app manifest configuration

Ang-m4 committed:
feat: create saleor order paid and updated webhooks

johanseto commented on platform_plugin_saleor/saleor_app/views.py:
```diff
@@ -0,0 +1,53 @@
+    """
+    payload = json.loads(request.body.decode("utf-8"))
+    token = payload.get("auth_token")
+    settings.SALEOR_API_TOKEN = token
```
This is too interesting. Anyway, I prefer to store that in the database somewhere like in application of django but let we consider it.

johanseto commented on platform_plugin_saleor/saleor_app/manifest.py:
```diff
@@ -0,0 +1,112 @@
+        dict: A dictionary containing the complete application manifest.
+    """
+
+    order_paid_query = """
```
I think queries would be in a folder of querys or somethinkg like that in other module

Ang-m4 committed:
feat: add order-created webhook with course enrollment

Ang-m4 committed:
feat: update product identifier

Ang-m4 committed:
refactor: redefine enroll course webhook

Ang-m4 committed:
feat: implement course enrollment pipeline for Saleor order

Ang-m4 commented:
@andrey-canon. I made a refactor of the web-hooks module from URLs to views logic. 

- Changed the enrollment trigger event to `ORDER_FULLY_PAID` that better match the business logic.
- Updated the Products information to add an `ExternalReference` as is easier to fetch the course-id that way.
- Added a simple `CourseEnrollment` pipeline to be executed as the order gets paid.
- Updated the `common.py` settings to include the pipeline steps.

johanseto commented on platform_plugin_saleor/manifest.py:
```diff
@@ -0,0 +1,68 @@
+        dict: A dictionary containing the complete application manifest.
+    """
+    manifest = {
+        'id': 'platform.plugin.saleor',
```
Could you make a constant for this?

Ang-m4 committed:
refactor: update fulfillment pipeline modules and names

Ang-m4 committed:
feat: implement order fulfillment functionality and warehouse retrieval

johanseto commented on platform_plugin_saleor/manifest.py:
```diff
@@ -0,0 +1,70 @@
+            'MANAGE_PRODUCT_TYPES_AND_ATTRIBUTES'
+        ],
+        'appUrl': f'{settings.LMS_ROOT_URL}',
+        'configurationUrl': f'{settings.LMS_ROOT_URL}/saleor/webhooks/configuration',
```
This view works?

Ang-m4 replied:
Taking a close look at the Apps configuration in [Saleor docs](https://docs.saleor.io/api-reference/apps/objects/app?), that field seems to be deprecated.

![image](https://github.com/user-attachments/assets/4841441a-0752-420a-bc21-031a9256b37e)

Ang-m4 committed:
refactor: remove deprecated manifest fields

johanseto reviewed (approved):
LGTM. Resolve conflict before merging
